src/translate.py

```diff
@@ -2016,6 +2013,16 @@ def get_meta(self, key: Register) -> Optional[RegMeta]:
         data = self.contents.get(key)
         return data.meta if data is not None else None
 
+    @contextmanager
+    def current_instr(self, instr: Instruction) -> Iterator[None]:
+        self.active_instr = instr
+        try:
+            yield
+        except Exception as e:
+            raise InstrProcessingFailure(instr) from e
```

Comment:
```suggestion
            with current_instr(instr):
                yield
```
to avoid code duplication?